hudhook/src/hooks/opengl3.rs

```diff
@@ -0,0 +1,285 @@
+
+    // Call the original function
+    trampoline_wglswapbuffers(dc)
+}
+
+static mut IMGUI_RENDER_LOOP: OnceCell<Box<dyn ImguiRenderLoop + Send + Sync>> = OnceCell::new();
+static mut IMGUI_RENDERER: Option<Mutex<Box<ImguiRenderer>>> = None;
+static TRAMPOLINE: OnceCell<OpenGl32wglSwapBuffers> = OnceCell::new();
+
+struct ImguiRenderer {
+    ctx: Context,
+    renderer: imgui_opengl_renderer::Renderer,
+    wnd_proc: WndProcType,
+    flags: ImguiRenderLoopFlags,
+    game_hwnd: HWND,
+}
+
+fn get_window_rect(hwnd: &HWND) -> Option<RECT> {
+    unsafe {
+        let mut rect: RECT = RECT { ..core::mem::zeroed() };
+        if GetWindowRect(*hwnd, &mut rect) != BOOL(0) {
+            Some(rect)
+        } else {
+            None
+        }
+    }
+}
+
+static mut LAST_FRAME: Option<Mutex<Instant>> = None;
+
+impl ImguiRenderer {
+    unsafe fn render(&mut self) {
+        if let Some(rect) = get_window_rect(&self.game_hwnd) {
+            let mut io = self.ctx.io_mut();
+            io.display_size = [(rect.right - rect.left) as f32, (rect.bottom - rect.top) as f32];
+            let mut pos = POINT { x: 0, y: 0 };
+
+            let active_window = GetForegroundWindow();
+            if !HANDLE(active_window.0).is_invalid()
+                && (active_window == self.game_hwnd
+                    || IsChild(active_window, self.game_hwnd).as_bool())
+            {
+                let gcp = GetCursorPos(&mut pos as *mut _);
+                if gcp.as_bool() && ScreenToClient(self.game_hwnd, &mut pos as *mut _).as_bool() {
+                    io.mouse_pos[0] = pos.x as _;
+                    io.mouse_pos[1] = pos.y as _;
+                }
+            }
+        } else {
+            trace!("GetWindowRect error: {:x}", GetLastError().0);
+        }
+
+        // Update the delta time of ImGui as to tell it how long has elapsed since the
+        // last frame
+        let last_frame = LAST_FRAME.get_or_insert_with(|| Mutex::new(Instant::now())).get_mut();
+        let now = Instant::now();
+        self.ctx.io_mut().update_delta_time(now.duration_since(*last_frame));
+        *last_frame = now;
+
+        let mut ui = self.ctx.frame();
+
+        IMGUI_RENDER_LOOP.get_mut().unwrap().render(&mut ui, &self.flags);
+        self.renderer.render(ui);
+    }
+
+    unsafe fn cleanup(&mut self) {}
+}
+
+impl ImguiWindowsEventHandler for ImguiRenderer {
+    fn io(&self) -> &imgui::Io {
+        self.ctx.io()
+    }
+
+    fn io_mut(&mut self) -> &mut imgui::Io {
+        self.ctx.io_mut()
+    }
+
+    fn focus(&self) -> bool {
+        self.flags.focused
+    }
+
+    fn focus_mut(&mut self) -> &mut bool {
+        &mut self.flags.focused
+    }
+
+    fn wnd_proc(&self) -> WndProcType {
+        self.wnd_proc
+    }
+}
+unsafe impl Send for ImguiRenderer {}
+unsafe impl Sync for ImguiRenderer {}
+
+/// Stores hook detours and implements the [`Hooks`] trait.
+pub struct OpenGL3Hooks {
```

Comment:
I'd rename this as `OpenGl3Hooks` to comply with the [API guidelines](https://rust-lang.github.io/api-guidelines/naming.html) that state:

> In UpperCamelCase, acronyms and contractions of compound words count as one word: use `Uuid` rather than `UUID`, `Usize` rather than `USize` or `Stdin` rather than `StdIn`.

Chances are this is a refactor that has to be done elsewhere in the codebase too, but it's worth it to do it as we go.

Reply:
Renamed